how-to/hints-and-tips/docs/visible-idle-detection.md

```diff
@@ -0,0 +1,97 @@
+> **_:information_source: OpenFin Workspace:_** [OpenFin Workspace](https://www.openfin.co/workspace/) is a commercial product and this repo is for evaluation purposes (See [LICENSE.MD](../LICENSE.MD)). Use of the OpenFin Container and OpenFin Workspace components is only granted pursuant to a license from OpenFin (see [manifest](../public/manifest.fin.json)). Please [**contact us**](https://www.openfin.co/workspace/poc/) if you would like to request a developer evaluation key or to discuss a production license.
+> OpenFin Workspace is currently **only supported on Windows** although you can run the sample on a Mac for development purposes.
+
+[<- Back to Table Of Contents](../README.md)
+
+# User Engagement - Idleness and Visibility
+
+If you are streaming data or are running an expensive web view then it can be beneficial to know:
+
+- Is my content currently visible or has the user switched tabs or minimized the window?
+- If my content is visible is the user currently idle or is the screen locked?
+
+## Checking to see if the user is idle or the screen is locked
+
+To enable this functionality you need the following permission: **idle-detection**
+
+You can add this to the permissions section of your manifest (either at the platform, window or view level):
+
+```js
+"permissions": {
+   "webAPIs": [
+    "idle-detection"
+    ]
+}
+```
+
+You should check to see if the permission is enabled in your code:
+
+```js
+const state = await IdleDetector.requestPermission();
+if (state !== 'granted') {
+  console.log('Idle detection permission not granted.');
+}
+```
+
+With the permission granted you can then add logic to be notified when the idle state changes:
+
+```js
+try {
+  const controller = new AbortController();
```

Comment:
Do you need the controller if you never abort it ?

Reply:
Added example of using it to dispose of listener